monai/data/utils.py

```diff
@@ -209,8 +217,11 @@ def iter_patch(
     # patches which are only in the padded regions
     iter_size = tuple(s + p for s, p in zip(arr.shape, patch_size_))
 
-    for slices in iter_patch_slices(iter_size, patch_size_, start_pos_padded):
-        yield arrpad[slices]
+    for coords in iter_patch_coordinates(iter_size, patch_size_, start_pos_padded):
+        slices = tuple(slice(coord[0], coord[1]) for coord in coords)
+        # compensate original image padding
+        coords_no_pad = tuple((coord[0] - p, coord[1] - p) for coord, p in zip(coords, patch_size_))
```

Comment:
If `iter_patch_coordinates` yielded `slice` objects as before you could pull out the `start` and `stop` members instead of `coords[0]` and `coords[1]`, this would be clearer about what's going on and not change the behaviour of `iter_patch_coordinates` from what it was.